mockhub.go

```diff
@@ -0,0 +1,205 @@
+package frontman
+
+// a mock hub used in tests, similar to https://bitbucket.org/cloudradar/debug_hub/
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"math/rand"
+	"net/http"
+	"strconv"
+)
+
+type MockHub struct {
+	address string
+}
+
+func NewMockHub(address string) *MockHub {
+	return &MockHub{
+		address: address,
+	}
+}
+
+func (hub *MockHub) URL() string {
+	return "http://" + hub.address
```

Comment:
why not provide full address in the address field?

Reply:
Because hostname and protocol are different things.

Reply:
maybe it's better to pass the separately but it's not good to hardcode protocol since it might be https or ftp(s)

Reply:
Yes but this mockhub.go has no external use case and is only used in testing. We may extend it if needed later on.